crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -13,17 +13,51 @@ use crate::{node_interner::StructId, Ident, Signedness};
 
 #[derive(Debug, PartialEq, Eq, Clone, Hash)]
 pub enum Type {
```

Comment:
It would be nice to cover the remaining variants of this enum. They're fairly self explanatory but we can give them a line each.

Reply:
Done